### PR TITLE
chore: handle exceptions with ControllerAdvice

### PR DESCRIPTION
**Pull request by NilsOveTen**

### CODE REVIEW

File: src/main/kotlin/no/digdir/service_catalog/configuration/CustomException.kt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package no.digdir.service_catalog.configuration
```
Comment (jeffreiffers) [inline src/main/kotlin/no/digdir/service_catalog/configuration/CustomException.kt:1]:
```suggestion
package no.digdir.servicecatalog.exception
```

File: src/main/kotlin/no/digdir/service_catalog/service/JsonPatchUtils.kt [hunk 2]
```diff
@@ -1,14 +1,13 @@
 package no.digdir.service_catalog.service
```
Comment (jeffreiffers) [inline src/main/kotlin/no/digdir/service_catalog/service/JsonPatchUtils.kt:1]:
```suggestion
package no.digdir.servicecatalog.service
```

File: src/main/kotlin/no/digdir/service_catalog/service/PublicServiceService.kt [hunk 3]
```diff
@@ -1,15 +1,13 @@
 package no.digdir.service_catalog.service
```
Comment (jeffreiffers) [inline src/main/kotlin/no/digdir/service_catalog/service/PublicServiceService.kt:1]:
```suggestion
package no.digdir.servicecatalog.service
```

**Review by jeffreiffers — CHANGES_REQUESTED** [review]

**Review by pooriamehregan — APPROVED** [review]

File: src/main/kotlin/no/digdir/service_catalog/configuration/CustomException.kt (hunk 1 above)
Comment (NilsOveTen) [reply src/main/kotlin/no/digdir/service_catalog/configuration/CustomException.kt]: 5cbff68
37a95bb

File: src/main/kotlin/no/digdir/service_catalog/service/JsonPatchUtils.kt (hunk 2 above)
Comment (NilsOveTen) [reply src/main/kotlin/no/digdir/service_catalog/service/JsonPatchUtils.kt]: 37a95bb

File: src/main/kotlin/no/digdir/service_catalog/service/PublicServiceService.kt (hunk 3 above)
Comment (NilsOveTen) [reply src/main/kotlin/no/digdir/service_catalog/service/PublicServiceService.kt]: 37a95bb

**Review by jeffreiffers — APPROVED** [review]

**Pull request merged**